src/aznfswatchdog

```diff
@@ -138,45 +140,96 @@ reconcile_conntrack()
     done
 }
 
-vecho "Starting aznfswatchdog..."
 
-# Detect and log distro and bash version
-if [ -f /etc/centos-release ]; then
-    linux_distro=$(cat /etc/centos-release 2>&1)
-elif [ -f /etc/os-release ]; then
-    linux_distro=$(grep "^PRETTY_NAME=" /etc/os-release | awk -F= '{print $2}' | tr -d '"')
-else
-    linux_distro="Unknown"
-fi
+#
+# Delete entry from $AZ_FILES_MOUNTMAP.
+#
+ensure_az_files_mountmap_not_exist()
+{
+    (
+        flock -e 999
```

Comment:
what is the purpose of a hardcoded fd?

Reply:
hardcoded fd is a random number we chode. There is no purpose as such. I followed v3's style and used hardcoded fd here.

Reply:
won't this line throw client failure? since if you execute this command directly in terminal, there will be failure, can you explain what is the purpose of this line and why this line works?

Reply:
flock -e is used to acquire exclusive lock on AZ_FILES_MOUNTMAP file. So if you run only flock -e 999, you will get bad descriptor error. You need to redirect 999 to some file. In this code, we are redirecting 999 to AZ_FILES_MOUNTMAP at line 194.
In ensure_az_files_mountmap_not_exist(), we are basically acquiring exclusive lock on AZ_FILES_MOUNTMAP file using flock -e command and doing some execution on AZ_FILES_MOUNTMAP. The lock will be released once the scope or ( ) parentheses ends.
What I meant by there is no purpose as such is there is no purpose to particularly choose 999 file descriptor. You can choose any random number (but in some range) like 123, 999 etc. But you need to redirect that file descriptor to some file else just flock -e <fd> won't succeed.